src/main/java/net/rubygrapefruit/platform/internal/jni/OsxFileEventFunctions.java

```diff
@@ -16,10 +16,55 @@
 
 package net.rubygrapefruit.platform.internal.jni;
 
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.NativeIntegration;
+import net.rubygrapefruit.platform.file.FileWatcher;
+import net.rubygrapefruit.platform.file.FileWatcherCallback;
 import net.rubygrapefruit.platform.internal.FunctionResult;
 
-public class OsxFileEventFunctions {
-    public static native void createWatch(String path, FunctionResult result);
-    public static native void startWatch(DefaultOsxFileEventFunctions.ChangeCallback callback, FunctionResult result);
-    public static native void stopWatch(FunctionResult result);
+import java.util.Collection;
+
+public class OsxFileEventFunctions implements NativeIntegration {
+
+    public FileWatcher startWatching(Collection<String> paths, double latency, FileWatcherCallback callback) {
```

Comment:
I would use an int or long and a `TimeUnit` to express the latency instead.

Reply:
Done.